lib/OperatingSystems/FreeBSD.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * @author Frank Karlitschek <frank@nextcloud.com>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ServerInfo\OperatingSystems;
+
+use bantu\IniGetWrapper\IniGetWrapper;
+
+/**
+ * Class FreeBSD
+ *
+ * @package OCA\ServerInfo\OperatingSystems
+ */
+class FreeBSD {
+
+	/** @var IniGetWrapper */
+	protected $phpIni;
+	
+	/**
+	 * FreeBSD constructor.
+	 *
+	 * @param IniGetWrapper $phpIni
+	 * @throws \Exception
+	 */
+	public function __construct(IniGetWrapper $phpIni) {
+		$this->phpIni = $phpIni;
+	}
+	
+	/**
+	 * @return bool
+	 */
+	public function supported(): bool {
+		return true;
+	}
+
+	/**
+	 * Get memory will return a list key => value where all values are in bytes.
+	 * [MemTotal => 0, MemFree => 0, MemAvailable => 0, SwapTotal => 0, SwapFree => 0].
+	 *
+	 * @return array
+	 */
+	public function getMemory(): array {
+		$data = ['MemTotal' => -1, 'MemFree' => -1, 'MemAvailable' => -1, 'SwapTotal' => -1, 'SwapFree' => -1];
+ 
+		$swapinfo = shell_exec('/usr/sbin/swapinfo');
```

Comment:
```suggestion
		try {
			$swapinfo = $this->executeCommand('/usr/sbin/swapinfo');
		} catch (\RuntimeException $e) {
			return $data;
		}
```

executeCommand is a method I added to DefaultOS as alternative to `is_function_enabled`. Please use it here for every execution of shell_exec and exec. It's actually a wrapper to makes it possible to mock/overwrite the function calls and write tests for the parsing logic.

Reply:
Done. Had to remove escapeshellcmd form executeCommand because of issues with piped commands

Reply:
Hmm. I would prefer to keep escapeshellcmd. What is done with pipes can also done with PHP.

Reply:
It might take a bit but I think I can do this.

Reply:
Thanks. If you need help paste the raw command output here and I can have a look how to parse it.

Reply:
serverinfo app is always released with Nextcloud hence we have some time until feature freeze for 20.

Reply:
Turns out regular expressions are not as hard as I thought it would. I have replaced all piped shell commands with the base commands and cleaned with regex.